unix_integration/common/src/client_sync.rs

```diff
@@ -83,64 +90,51 @@ impl DaemonClientBlocking {
             .map_err(Box::new)?;
 
         // Now wait on the response.
+        data.clear();
         let start = SystemTime::now();
         let mut read_started = false;
-        let mut data = Vec::with_capacity(1024);
-        let mut counter = 0;
 
         loop {
-            let mut buffer = [0; 1024];
             let durr = SystemTime::now().duration_since(start).map_err(Box::new)?;
             if durr > timeout {
                 error!("Socket timeout");
                 // timed out, not enough activity.
-                break;
+                return Err(Box::new(io::Error::other("Timeout")));
             }
+
+            let mut buffer = [0; 1024];
+
             // Would be a lot easier if we had peek ...
             // https://github.com/rust-lang/rust/issues/76923
             match self.stream.read(&mut buffer) {
                 Ok(0) => {
                     if read_started {
-                        debug!("read_started true, we have completed");
+                        debug!("read_started true, no bytes read");
                         // We're done, no more bytes.
-                        break;
+                        // break;
```

Comment:
Is this deliberate (it being commented out rather than removed)? If so, I'm curious as to why